### PR TITLE
Only support copy for query templates

### PR DESCRIPTION
**Pull request by amsiglan**
### Description
For query templates we don't want to run them when they load because they need user input to be valid. Currently Discover does not support a good way to avoid running the query on load.
To avoid this, this PR only lets the user copy the query, which can then be pastes into the Discover editor for modification before running them.

## Screenshot

https://github.com/user-attachments/assets/52508d7a-a9fd-4c8e-b4d0-1c3bc0f159c4

## Testing the changes
Tested changes locally

## Changelog
- fix: Only support copy action for query templates

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: src/plugins/data/public/ui/saved_query_flyouts/open_saved_query_flyout.tsx [hunk 1]
```diff
@@ -301,25 +343,12 @@ export function OpenSavedQueryFlyout({
           </EuiFlexItem>
           <EuiFlexItem grow={false}>
             <EuiButton
-              disabled={!selectedQuery}
+              // disabled={!selectedQuery}
```
Comment (virajsanghvi) [inline src/plugins/data/public/ui/saved_query_flyouts/open_saved_query_flyout.tsx:346]: not blocking, but can we clean up dead code?

**Review by virajsanghvi — APPROVED** [review]
code looks safe to me although I think it'll end up failing tests for the flyout (you may want to run locally to get ahead of it) - approving to unblock and assuming it passes ci 

let's make sure we follow-up with testing if it does pass

File: src/plugins/data/public/ui/saved_query_flyouts/open_saved_query_flyout.tsx (hunk 1 above)
Comment (amsiglan) [reply src/plugins/data/public/ui/saved_query_flyouts/open_saved_query_flyout.tsx]: Fixed

File: src/plugins/data/public/ui/saved_query_flyouts/open_saved_query_flyout.tsx [hunk 2]
```diff
@@ -111,6 +122,7 @@ export function OpenSavedQueryFlyout({
     fetchAllSavedQueriesForSelectedTab();
     setSearchQuery(EuiSearchBar.Query.MATCH_ALL);
     updatePageIndex(0);
+    setSelectedQuery(undefined);
```
Comment (virajsanghvi) [inline src/plugins/data/public/ui/saved_query_flyouts/open_saved_query_flyout.tsx:125]: curious, why do you have to do this?

Comment (virajsanghvi) [reply src/plugins/data/public/ui/saved_query_flyouts/open_saved_query_flyout.tsx:125]: nevermind, assuming you want to unset when reloading...

**Review by virajsanghvi — APPROVED** [review]

File: src/plugins/data/public/ui/saved_query_flyouts/open_saved_query_flyout.tsx (hunk 2 above)
Comment (amsiglan) [reply src/plugins/data/public/ui/saved_query_flyouts/open_saved_query_flyout.tsx:125]: Yes when the queries are reloaded or tab is changed

**Review by ashwin-pc — APPROVED** [review]

**Review by huyaboo — APPROVED** [review]

**Pull request merged**